knowledge-base/contextmenu-treeview-items.md

```diff
@@ -0,0 +1,147 @@
+---
+title: Add ContextMenu to the TreeView items
+description: How add ContextMenu to the TreeView items.
+type: how-to
+page_title: Add ContextMenu to the TreeView items
+slug: contextmenu-kb-treeview-item
+position: 
+tags: 
+ticketid: 1508692
+res_type: kb
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product</td>
+			<td>ContextMenu for Blazor, TreeView for Blazor</td>
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+
+I would like to add the ContextMenu component to every item (node) in the TreeView.
+
+## Solution
+
+The ContextMenu exposes an API to associate the component to any DOM element through the [ShowAsync]({%slug contextmenu-integration%}) method. You can use the [oncontextmenu](https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/oncontextmenu) event of an HTML element in the [treeview item template]({%slug components/treeview/templates%}) to show the context menu by using the `MouseEventArgs`. 
+
+
+
+````CSHTML
+@* Use the oncontextmenu event of the HTML element to show the ContextMenu for the TreeView items *@
+
+LastClickedItem = item;
```

Comment:
The code snippet seems to be cut off, all the markup is missing.

Reply:
Done, thank you, my bad